ui/src/app/shared/workflow-actions.ts

```diff
@@ -0,0 +1,64 @@
+import {NodePhase, Workflow} from '../../models';
+import {services} from './services';
+import {Utils} from './utils';
+
+export type ActionDisabled = {
+    [action in WorkflowActionName]: boolean;
+};
+
+export type WorkflowActionName = 'RETRY' | 'RESUBMIT' | 'SUSPEND' | 'RESUME' | 'STOP' | 'TERMINATE' | 'DELETE';
```

Comment:
the actions map to API which use lowercase names?

Reply:
also - I think you should use the term `operation` rather than `action`

Reply:
I had them as lowercase but changed to upper for labeling purposes. Simon pointed out that the way I was initially converting lowercase to a label (`string.charAt(0).toUpperCase() + string.slice(1);`) was clunky and just having them as uppercase would be easier. Thinking about it now though, it wouldn't be clunky to have them lowercase and do `string.toUpperCase()` to convert. The union type is never used to actually query the API here, so would the change to lowercase be for consistency? 

Operation sounds good, I'll change that.